### PR TITLE
use XO for linting instead of ESLint

### PR DESCRIPTION
**Pull request by lance**
Connects to: https://github.com/bucharest-gold/opossum/issues/175

### CODE REVIEW

File: lib/hystrix-stats.js [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
+// Use a single hystrix stream for all circuits
 const hystrixStream = new Transform({
   objectMode: true,
   transform (stats, encoding, cb) {
     return cb(null, `data: ${JSON.stringify(formatter(stats))}\n\n`);
   }
 });
 
-// TODO: This number is somewhat arbitrary. However, we need to allow
+// This number is somewhat arbitrary. Howver, we need to allow
```
Comment (helio-frota) [inline lib/hystrix-stats.js]: A typo

File: lib/semaphore.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
-module.exports = exports = semaphore;
+exports = semaphore;
+module.exports = exports;
 
 function semaphore (count) {
   const resolvers = [];
   let counter = count;
 
-  let sem = {
+  const sem = {
```
Comment (helio-frota) [inline lib/semaphore.js]: good that xo catched this

File: lib/status.js [hunk 3]
```diff
@@ -66,29 +66,37 @@ class Status extends EventEmitter {
...
 
-    // prime the window with buckets
-    for (let i = 0; i < this[BUCKETS]; i++) this[WINDOW][i] = bucket();
+    // Prime the window with buckets
+    for (let i = 0; i < this[BUCKETS]; i++) {
+      this[WINDOW][i] = bucket();
+    }
 
-    // rotate the buckets periodically
+    // Potate the buckets periodically
```
Comment (helio-frota) [inline lib/status.js]: typo

File: package.json [hunk 4]
```diff
@@ -20,7 +20,33 @@
     "postpublish": "./publish-docs.sh",
     "prerelease": "npm test",
     "release": "standard-version -s",
-    "lint": "eslint examples/*/*.js test/*.js index.js lib/*.js"
+    "lint": "npx xo index.js lib/*.js test/*.js"
```
Comment (lholmquist) [inline package.json]: do you need `npx` here?  and i don't think you need to specify the files you want to lint, i believe by default, xo will lint any `.js` file in the whole project

Comment (lance) [reply package.json]: Yeah, but I didn't want to include the `examples` directory for now. It's catching a LOT of stuff in the React bits that I wasn't sure how best to deal with.

**Comment by lance** [comment]
Closing this as discussed. We should revisit our use of eslint, but creating another dependency on top of it is maybe not going to be our best approach.

**Pull request closed without merging**